### PR TITLE
refactor : get rid of LangServerCtx

### PR DESCRIPTION
**Pull request by ap891843**
refactor : get rid of LangServerCtx

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

### CODE REVIEW

File: server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java [hunk 1]
```diff
@@ -14,20 +14,29 @@
...
+import static java.util.Optional.ofNullable;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public abstract class ConfigurableTest {
-  @BeforeAll
-  public static void setUp() {
-    LangServerCtx.getGuiceCtx(new TestModule(), new EngineModule(), new DatabusModule());
-  }
+  protected static final String PATH_TO_TEST_RESOURCES = "filesToTestPath";
+  protected Injector injector =
```
Comment (temanbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java:31]: Why is it protected?

File: server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java [hunk 2]
```diff
@@ -14,20 +14,29 @@
...
-  public static void tearDown() {
-    LangServerCtx.shutdown();
+  /**
+   * Retrieve {@link CobolTextRegistry} using file-based implementation.
+   *
+   * @return {@link CobolTextRegistry}.
+   */
+  public static CobolTextRegistry retrieveTextsRegistry() {
+    return new FolderTextRegistry(ofNullable(getProperty(PATH_TO_TEST_RESOURCES)).orElse(""));
   }
```
Comment (temanbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java:41]: Seems like it may be encapsulated in `PositiveTest`

**Comment by temanbrcom** [comment]
@grianbrcom 
@asatklichov 
@Nurkambay

File: server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java (hunk 1 above)
Comment (ap891843) [reply server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java:31]: Yes, correct need not be. Private would suffice.

File: server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java (hunk 2 above)
Comment (ap891843) [reply server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java:41]: We need this for negativeTest as well

**Review by temanbrcom — APPROVED** [review]
🖌️

File: server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java (hunk 1 above)
Comment (ap891843) [reply server/src/test/java/com/broadcom/lsp/cobol/ConfigurableTest.java:31]: resolved on chat.

**Review by grianbrcom — APPROVED** [review]
:broom:

File: server/src/test/java/com/broadcom/lsp/cobol/positive/PositiveTest.java [hunk 3]
```diff
@@ -47,10 +44,7 @@
    * @return a collection of objects that would be passed to the constructor one by one.
    */
   private static Stream<CobolText> retrieveTextsToTest() {
-    CobolTextRegistry registry =
-        new FolderTextRegistry(ofNullable(getProperty(PATH_TO_TEST_RESOURCES)).orElse(""));
-    copybooks = registry.getCopybooks();
```
Comment (temanbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/positive/PositiveTest.java:52]: `copybooks` not initialized

**Review by temanbrcom — CHANGES_REQUESTED** [review]

File: server/src/test/java/com/broadcom/lsp/cobol/positive/PositiveTest.java (hunk 3 above)
Comment (ap891843) [reply server/src/test/java/com/broadcom/lsp/cobol/positive/PositiveTest.java:52]: Great catch!
Done.

**Review by temanbrcom — APPROVED** [review]

**Review by asatklichov — APPROVED** [review]
🧇

**Pull request merged**